src/components/ReportPage/index.jsx

```diff
@@ -158,6 +177,39 @@ class ReportPage extends PureComponent {
     return formattedDate;
   }
 
+    handlePagination = (action) => {
+      const { pagination: { currentPage, numberOfPages, limit } } = this.state;
+      const page = pageNavigation(action, currentPage, numberOfPages);
+      this.reportData(page, limit);
+    }
+
+    setPaginationState = (pagination) => {
+      this.setState(prevState => ({
+        ...prevState,
+        pagination: {
+          ...prevState.pagination,
+          ...pagination,
+        },
+      }));
+    }
+
+  onChangePage = (event) => {
+    const { pagination: { numberOfPages, limit } } = this.state;
+    event.preventDefault();
+    const currentPage = pageChange(event.target.value, numberOfPages);
+    this.setPaginationState({ currentPage });
+    setTimeout(() => { this.reportData(currentPage, limit); }, 1500);
```

Comment:
Why the use of setTimeout while the time taken to go to the next page depends on server response speed. At times it may be fast at time slower than 1500ms.

Reply:
I'm using setTimeout here to allow the user to type in the page before I make a call to the back end. Without this, the app will be making a backend call each time a user presses a number on the keyboard

Reply:
Oh. Yeah. That makes sense.